modules/services/mbsync.nix

```diff
@@ -75,35 +85,47 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
-    assertions = [
-      (lib.hm.assertions.assertPlatform "services.mbsync" pkgs
-        lib.platforms.linux)
-    ];
-
-    systemd.user.services.mbsync = {
-      Unit = { Description = "mbsync mailbox synchronization"; };
-
-      Service = {
-        Type = "oneshot";
-        ExecStart =
-          "${cfg.package}/bin/mbsync ${concatStringsSep " " mbsyncOptions}";
-      } // (optionalAttrs (cfg.postExec != null) {
-        ExecStartPost = cfg.postExec;
-      }) // (optionalAttrs (cfg.preExec != null) {
-        ExecStartPre = cfg.preExec;
-      });
-    };
+  config = mkIf cfg.enable (mkMerge [
+    (optionalAttrs (hasAttr "sytemd" options) {
```

Comment:
Probably better to use `pkgs.stdenv.hostPlatform.system` like `lib.hm.assertions.assertPlatform` does it.

Reply:
So is something like `(optionalAttrs (elem pkgs.stdenv.hostPlatform lib.platforms.linux) {` what you mean? Because that doesn't compile due to infinite recursion. I think it's because you can't check the platform on the top level, which is why I used the `hasAttr` pattern originally.

Reply:
The trick is to almost never use `optionalAttrs` when dealing with NixOS modules. It is a sure-fire way to end up in infinite recursion. You'll want to use

    mkIf pkgs.stdenv.hostPlatform.isDarwin { … }

instead. For example

https://github.com/nix-community/home-manager/blob/684e85d01d333be91c4875baebb05b93c7d2ffaa/modules/targets/darwin/fonts.nix#L15